pkg/operation/shoot/shoot.go

```diff
@@ -498,12 +472,9 @@ func (s *Shoot) ComputeOutOfClusterAPIServerAddress(apiServerAddress string, use
 
 // IPVSEnabled returns true if IPVS is enabled for the shoot.
 func (s *Shoot) IPVSEnabled() bool {
-	s.infoMutex.RLock()
-	defer s.infoMutex.RUnlock()
-
-	return s.info.Spec.Kubernetes.KubeProxy != nil &&
-		s.info.Spec.Kubernetes.KubeProxy.Mode != nil &&
-		*s.info.Spec.Kubernetes.KubeProxy.Mode == gardencorev1beta1.ProxyModeIPVS
+	return s.GetInfo().Spec.Kubernetes.KubeProxy != nil &&
+		s.GetInfo().Spec.Kubernetes.KubeProxy.Mode != nil &&
+		*s.GetInfo().Spec.Kubernetes.KubeProxy.Mode == gardencorev1beta1.ProxyModeIPVS
```

Comment:
This can lead to an inconsistent result if info is updated in parallel. I think we should either return to using `RLock` or work with a local variable like `info := s.GetInfo()` in this function. The same is true for other functions in this file which call `s.GetInfo()` multiple times in a row.

WDYT?

Reply:
I think `RLock` should not be used anymore, since with the latest changes it's conceptually redundant. I like the second suggestion, let me double check and fix this where needed.

Reply:
As discussed, did this in a handful of places (mostly helper methods), but refrained from changing that in the rest of the code base. The main reason is that most calls to `GetInfo` are reading stuff we are not updating anyway. Only the user could perform such updates in parallel to the flow, but we are anyway not adequately protected from this and just avoiding muliple calls to `GetInfo` won't save us.